import "google/protobuf/descriptor.proto";

package a4.io;

enum MetadataMergeOptions {
    MERGE_BLOCK_IF_DIFFERENT = 0; // Throw exception if different, default
    MERGE_ADD = 1;      // Add numeric values or strings together
    MERGE_MULTIPLY = 2; // Multiply numeric values together
    MERGE_AND = 2;      // Take the AND on booleans
    MERGE_UNION = 3;    // Concatenate repeated fields (no duplicates)
    MERGE_APPEND = 4;   // Concatenate repeated fields
    MERGE_DROP = 5;     // Drop this field on merging
}

extend google.protobuf.FieldOptions {
    optional MetadataMergeOptions merge = 50002;
}

extend google.protobuf.MessageOptions {
    optional int32 fixed_class_id = 50000;
    optional string major_version = 50010;
    optional string minor_version = 50011;
    optional string patch_version = 50012;
}

